src/entities/BaseEntity.java

```diff
@@ -0,0 +1,63 @@
+package entities;
+
+import java.time.ZonedDateTime;
+
+public abstract class BaseEntity {
+    private Long id;
```

Comment:
What would happen if the id is not Long?

Reply:
In the project, I'll make all the id of the entities of the project be of type Long
Are you refering to what will happen if in the future i decide to change the variable type of the id's?

Reply:
I don't think you would like to have Long ids to solve security related issues

Reply:
Ok, i'll change the id to make it more flexible